### ISSUE TITLE
Custom JWT payload

### ISSUE THREAD

**Issue by kpn13** [bshaffer/oauth2-server-php#750]
Hi,

Thank you for this lib, it has save me a lot of time by now !

I'm facing a problem to custom the payload of a JWT. I think I have to override \OAuth2\ResponseType\JwtAccessToken, but I didn't find a way.

I gave my own implementation of JwtAccessToken in my Server.php in responseType field but it didn't work.

I think the problem could be createDefaultJwtAccessTokenResponseType() in OAuth2\Server.php return a default instance of JwtAccessToken without check if there is a custom JwtAccessToken in the responseTypes array. Are you agree with thas or am I totally wrong ?

Could you help me with that please ?

Thank you in advance.

**Comment by afilippov1985** [comment]
Override `OAuth2\ResponseType\JwtAccessToken::createAccessToken()` or create your own class from scratch (it must implement `OAuth2\ResponseType\AccessTokenInterface`)
Then initialize `OAuth2\Server` passing 4th parameter
```php
$responseTypes = array('token' => new MyCustomJwtAccessToken(............));
$server = new OAuth2\Server($storage, $config, $grantTypes, $responseTypes);
```

**Comment by achimha** [comment]
I believe it is a very common use case to add additional fields to a JWT. The other libraries I have dealt with provide simple ways to add user claims.

While I can override the `JwtAccessToken` class, I have to provide a complete implementation, copying the existing logic because there is no way to add extra fields.

**Comment by galves** [comment]
You should be at least able to set the claims outlined here https://www.iana.org/assignments/jwt/jwt.xhtml

**Comment by svycka** [comment]
will be easier when this will be released https://github.com/bshaffer/oauth2-server-php/pull/795

**Comment by bshaffer** [comment]
This is available now in `v1.10`. See [the CHANGELOG](https://github.com/bshaffer/oauth2-server-php/blob/master/CHANGELOG.md) for details!
